data_model/src/block.rs

```diff
@@ -278,6 +278,40 @@ impl SignedBlock {
             SignatureOf::new(private_key, &block.payload),
         ));
     }
+
+    /// Creates genesis block without signatures (not signed by any peer)
+    pub fn genesis(genesis_transaction: SignedTransaction, topology: Vec<PeerId>) -> SignedBlock {
+        let transactions_hash = vec![genesis_transaction.hash()]
+            .into_iter()
+            .collect::<MerkleTree<_>>()
+            .hash()
+            .expect("Tree is not empty");
+        let timestamp_ms = u64::try_from(genesis_transaction.creation_time().as_millis())
+            .expect("Must fit since Duration was created from u64 in creation_time()");
+        let header = BlockHeader {
+            height: 1,
+            prev_block_hash: None,
+            transactions_hash,
+            timestamp_ms,
+            view_change_index: 0,
+            consensus_estimation_ms: 0,
+        };
+        let transaction = CommittedTransaction {
+            value: genesis_transaction,
+            error: None,
+        };
+        let payload = BlockPayload {
+            header,
+            commit_topology: topology,
+            transactions: vec![transaction],
+            event_recommendations: vec![],
+        };
+        SignedBlockV1 {
+            signatures: vec![],
```

Comment:
Shouldn't we include genesis signature here?
So that other peers can verify that genesis block was indeed signed by genesis. 
I think this is not checked right now only on transaction level, so sender can easily replace other fields.

Reply:
Changed as you suggested. By the way there are plans to use genesis hash instead of signing (#4555)